### PR TITLE
Implement transaction processor for NFT module

### LINKED ISSUE #1795: Implement transaction processor for NFT module
### Description
To index transactions, we need to implement a transaction processor for the NFT module

### Acceptance Criteria

- Implement processor for NFT module in indexer
- Index NFT module transactions
- Index/update NFT table with the information from transactions
- Add unit tests if applicable

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #1795 

### How was it solved?
- [x] Implement processor for NFT module in indexer
- [x] Index NFT module transactions
- [x] Index/update NFT table with the information from transactions
- [x] Add unit tests if applicable (N/A)

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
+	nft.owner = tx.recipientAddress;
+	await nftTable.upsert(nft, dbTrx);
+	logger.trace(`Updated owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js:62]:
```suggestion
	logger.debug(`Updated owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
```

File: services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
+	nft.owner = tx.senderAddress;
+	await nftTable.upsert(nft, dbTrx);
+	logger.trace(`Updated owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js:92]:
```suggestion
	logger.debug(`Updated owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
```

File: services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	};
+
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
+	nft.owner = tx.recipientAddress;
+	await nftTable.upsert(nft, dbTrx);
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js:61]: I think there is no need to call `db.find` here. Instead we can directly `upsert`
```suggestion
	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
	const nft = {
	   nftID: tx.nftID,
	   owner: tx.recipientAddress,
	};
	await nftTable.upsert(nft, dbTrx);
```

File: services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		...tx.params,
+	};
+
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
+	nft.owner = tx.senderAddress;
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js:90]: Same here

File: services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transferCrossChain.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
+	nft.owner = tx.recipientAddress;
+	nft.escrowChainID = tx.receivingChainID;
+	await nftTable.upsert(nft, dbTrx);
+	logger.trace(`Updated owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transferCrossChain.js:63]:
```suggestion
	logger.debug(`Updated owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
```

File: services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transferCrossChain.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
+	nft.owner = tx.senderAddress;
+	nft.escrowChainID = null;
+	await nftTable.upsert(nft, dbTrx);
+	logger.trace(`Updated owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transferCrossChain.js:94]:
```suggestion
	logger.debug(`Updated owner of nft ${tx.nftID} to ${tx.senderAddress}.`);
```

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**